Tests/NumLib/LocalToGlobalIndexMapMultiComponent.cpp

```diff
@@ -200,8 +234,68 @@ void NumLibLocalToGlobalIndexMapMultiDOFTest::test(
     }
 }
 
+// Multicomponent version
+template <NL::ComponentOrder ComponentOrder>
+void NumLibLocalToGlobalIndexMapMultiDOFTest::test(
+    unsigned const num_components,
+    std::vector<int> const& selected_components,
+    std::function<std::size_t(std::size_t, std::size_t)> const&
+        compute_global_index)
+{
+    initComponents(num_components, selected_components, ComponentOrder);
 
+    ASSERT_EQ(dof_map->getNumberOfComponents(), num_components);
+    ASSERT_EQ(dof_map->size(), mesh->getNumberOfElements());
 
+    ASSERT_EQ(dof_map_boundary->getNumberOfComponents(),
+              selected_components.size());
+    ASSERT_EQ(dof_map_boundary->size(), boundary_elements.size());
+
+    // check mesh elements
+    for (unsigned e = 0; e < dof_map->size(); ++e)
+    {
+        auto const element_nodes_size = mesh->getElement(e)->getNumberOfNodes();
+        auto const ptr_element_nodes = mesh->getElement(e)->getNodes();
+
+        for (unsigned c = 0; c < dof_map->getNumberOfComponents(); ++c)
+        {
+            auto const& global_idcs = (*dof_map)(e, c).rows;
+            ASSERT_EQ(element_nodes_size, global_idcs.size());
+
+            for (unsigned n = 0; n < element_nodes_size; ++n)
+            {
+                auto const node_id = ptr_element_nodes[n]->getID();
+                auto const glob_idx = compute_global_index(node_id, c);
+                EXPECT_EQ(glob_idx, global_idcs[n]);
+            }
+        }
+    }
+
+    // check boundary elements
+    for (unsigned e = 0; e < dof_map_boundary->size(); ++e)
+    {
+        ASSERT_EQ(selected_components.size(),
+                  dof_map_boundary->getNumberOfComponents());
+
+        for (unsigned c = 0; c < selected_components.size(); ++c)
+        {
+            auto const& global_idcs = (*dof_map_boundary)(e, c).rows;
+
+            ASSERT_EQ(
+                2,
+                global_idcs.size());  // boundary of quad is line with two nodes
+
+            for (unsigned n = 0; n < 2;
+                 ++n)  // boundary of quad is line with two nodes
+            {
+                auto const node = e + n;
```

Comment:
I already forgot it. The multi-component test-code is copy of the single-component with an additional loop over the components.
Let's try to figure out: e is the number of a boundary element (which is a line, so two nodes) from 0 to something. e+n must be the node ids along the x-axis of the mesh; I changed the boundary to be in y direction, and the test failed, so the assumption seems reasonable.